front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/ChartWidgets/ParticipationWidget/useParticipation/parse/parseTimeSeries.ts

```diff
@@ -0,0 +1,158 @@
+import moment, { Moment } from 'moment';
+
+import {
+  ParticipationResponse,
+  TimeSeriesResponseRow,
+} from 'api/graph_data_units/responseTypes/ParticipationWidget';
+
+import {
+  timeSeriesParser,
+  getFirstDateInData,
+  getLastDateInData,
+} from 'components/admin/GraphCards/_utils/timeSeries';
+import { IResolution } from 'components/admin/ResolutionControl';
+
+import { get } from 'utils/helperUtils';
+
+import { CombinedTimeSeriesRow, GenericTimeSeriesRow } from '../../typings';
+
+const getEmptyRow = (date: Moment) => ({
+  date: date.format('YYYY-MM-DD'),
+  count: 0,
+});
+
+const parseRow = (
+  date: Moment,
+  row?: TimeSeriesResponseRow
+): GenericTimeSeriesRow => {
+  if (!row) return getEmptyRow(date);
+
+  return {
+    count: row.count,
+    date: date.format('YYYY-MM-DD'),
+  };
+};
+
+const getDate = (row: TimeSeriesResponseRow) => {
+  return moment(get(row, 'first_dimension_date_created_date'));
+};
+
+const _parseTimeSeries = timeSeriesParser(getDate, parseRow);
```

Comment:
why `_` would it not be better to name it something else? However, not sure I'm understanding how this is working and why?